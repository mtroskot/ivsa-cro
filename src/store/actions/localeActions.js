import { localeActionTypes } from 'src/constants/actionTypes';

export function changeLocale(locale) {
  return {
    type: localeActionTypes.CHANGE_LOCALE,
    payload: { locale }
  };
}
export function changeLocaleSuccess(locale) {
  return {
    type: localeActionTypes.CHANGE_LOCALE_SUCCESS,
    payload: { locale }
  };
}
