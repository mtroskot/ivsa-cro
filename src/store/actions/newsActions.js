import { newsActionTypes } from 'src/constants/actionTypes';

export function fetchNews(refreshing = false) {
  return {
    type: newsActionTypes.FETCH_NEWS,
    payload: { refreshing }
  };
}
export function fetchNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.FETCH_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}
export function fetchNewsError(error) {
  return {
    type: newsActionTypes.FETCH_NEWS_ERROR,
    payload: { error }
  };
}

export function publishNews(date, publishedBy, notice) {
  return {
    type: newsActionTypes.PUBLISH_NEWS,
    payload: { date, publishedBy, notice }
  };
}

export function publishNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.PUBLISH_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}

export function updateNews(newsId, date, publishedBy, notice) {
  return {
    type: newsActionTypes.UPDATE_NEWS,
    payload: { newsId, date, publishedBy, notice }
  };
}

export function updateNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.UPDATE_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}

export function deleteNews(newsId) {
  return {
    type: newsActionTypes.DELETE_NEWS,
    payload: { newsId }
  };
}

export function deleteNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.DELETE_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}
