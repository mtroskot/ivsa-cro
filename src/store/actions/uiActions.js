import { uiActionTypes } from 'src/constants/actionTypes';
import { ObjectUtils } from 'src/utils';

export const startAction = (name, params) => {
  if (!ObjectUtils.isObject(params)) {
    throw Error(`type of ${typeof params} is not an object`);
  }
  return {
    type: uiActionTypes.START_ACTION,
    payload: {
      action: {
        name,
        params
      }
    }
  };
};

export const stopAction = name => ({
  type: uiActionTypes.STOP_ACTION,
  payload: { name }
});

export const refreshActionStart = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_START,
  payload: { refreshAction }
});

export const refreshActionStop = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_STOP,
  payload: { refreshAction }
});

export const togglePopupMessage = (message, position) => ({
  type: uiActionTypes.TOGGLE_POPUP_MESSAGE,
  payload: {
    message,
    position
  }
});
