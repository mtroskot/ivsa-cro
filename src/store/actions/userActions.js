import { userActionTypes } from 'src/constants/actionTypes';

export function authenticateUser(email, password) {
  return {
    type: userActionTypes.AUTH_USER,
    payload: { email, password }
  };
}

export const authenticateUserSuccess = (displayName, email, idToken, expiryDate) => {
  return {
    type: userActionTypes.AUTH_USER_SUCCESS,
    payload: { displayName, email, idToken, expiryDate }
  };
};

export const authenticateUserError = error => {
  return {
    type: userActionTypes.AUTH_USER_ERROR,
    payload: { error }
  };
};

export const logout = () => {
  return {
    type: userActionTypes.LOG_OUT,
    payload: {}
  };
};
