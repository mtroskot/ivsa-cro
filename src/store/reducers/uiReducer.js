import { uiActionTypes } from 'src/constants/actionTypes';

export const initialState = {
  loader: {
    actions: [],
    refreshing: []
  },
  popupMessage: {
    message: null,
    position: null
  }
};

const uiReducer = (state = initialState, { type, payload }) => {
  const { loader } = state;
  const { actions, refreshing } = loader;
  switch (type) {
    case uiActionTypes.START_ACTION:
      return {
        ...state,
        loader: {
          ...loader,
          actions: [...actions, payload.action]
        }
      };
    case uiActionTypes.STOP_ACTION:
      return {
        ...state,
        loader: {
          ...loader,
          actions: actions.filter(action => action.name !== payload.name)
        }
      };
    case uiActionTypes.REFRESH_ACTION_START:
      return {
        ...state,
        loader: {
          ...loader,
          refreshing: [...refreshing, payload.refreshAction]
        }
      };
    case uiActionTypes.REFRESH_ACTION_STOP:
      return {
        ...state,
        loader: {
          ...loader,
          refreshing: refreshing.filter(refresh => refresh !== payload.refreshAction)
        }
      };
    case uiActionTypes.TOGGLE_POPUP_MESSAGE:
      return {
        ...state,
        popupMessage: {
          message: payload.message,
          position: payload.position
        }
      };
    default:
      return state;
  }
};

export default uiReducer;
