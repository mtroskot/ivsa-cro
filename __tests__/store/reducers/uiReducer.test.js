import uiReducer,{initialState} from 'src/store/reducers/uiReducer';
import * as uiActions from 'src/store/actions/uiActions';

jest.mock('src/utils', () => ({
  ObjectUtils: { isObject: jest.fn().mockReturnValue(true) }
}));

describe('INITIAL_STATE', () => {
  const beforeActionState = initialState;

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState
    expect(uiReducer(beforeActionState, action)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns start action state', () => {
    const name = 'ACTION_NAME';
    const startAction = uiActions.startAction(name);
    const expectedState = {
      loader: {
        actions: [startAction.payload.action],
        refreshing: []
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns stop action state', () => {
    const name = 'ACTION_NAME';
    const stopAction = uiActions.stopAction(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: []
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, stopAction)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, stopAction)).toMatchSnapshot();
  });

  it('4returns refresh action start state', () => {
    const name = 'ACTION_NAME';
    const refreshActionStart = uiActions.refreshActionStart(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: [refreshActionStart.payload.refreshAction]
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, refreshActionStart)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, refreshActionStart)).toMatchSnapshot();
  });

  it('5returns refresh action stop state', () => {
    const name = 'ACTION_NAME';
    const refreshActionStop = uiActions.refreshActionStop(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: []
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, refreshActionStop)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, refreshActionStop)).toMatchSnapshot();
  });

  it('6returns toggle popup message state', () => {
    const message = 'message';
    const position = 'position';
    const togglePopupMessage = uiActions.togglePopupMessage(message, position);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: []
      },
      popupMessage: {
        message: message,
        position: position
      }
    };
    expect(uiReducer(beforeActionState, togglePopupMessage)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, togglePopupMessage)).toMatchSnapshot();
  });
});

describe('START_ACTION_STATE', () => {
  const beforeActionState = {
    ...initialState,
    loader: {
      actions: [{ name: 'ACTION_NAME' }, { name: 'ACTION_NAME' }, { name: 'ACTION_NAME2' }],
      refreshing: []
    }
  };

  it('7returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(uiReducer(beforeActionState, action)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('8returns start action state', () => {
    const name = 'ACTION_NAME';
    const startAction = uiActions.startAction(name);
    const expectedState = {
      loader: {
        actions: [...beforeActionState.loader.actions, startAction.payload.action],
        refreshing: []
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('9returns stop action state', () => {
    const name = 'ACTION_NAME';
    const stopAction = uiActions.stopAction(name);
    const expectedState = {
      loader: {
        actions: [{ name: 'ACTION_NAME2' }],
        refreshing: []
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, stopAction)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, stopAction)).toMatchSnapshot();
  });

  it('10returns refresh action start state', () => {
    const name = 'ACTION_NAME';
    const refreshActionStart = uiActions.refreshActionStart(name);
    const expectedState = {
      loader: {
        actions: beforeActionState.loader.actions,
        refreshing: [refreshActionStart.payload.refreshAction]
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, refreshActionStart)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, refreshActionStart)).toMatchSnapshot();
  });

  it('11returns refresh action stop state', () => {
    const name = 'ACTION_NAME';
    const refreshActionStop = uiActions.refreshActionStop(name);
    const expectedState = {
      loader: {
        actions: beforeActionState.loader.actions,
        refreshing: []
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, refreshActionStop)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, refreshActionStop)).toMatchSnapshot();
  });

  it('12returns toggle popup message state', () => {
    const message = 'message';
    const position = 'position';
    const togglePopupMessage = uiActions.togglePopupMessage(message, position);
    const expectedState = {
      loader: {
        actions: beforeActionState.loader.actions,
        refreshing: []
      },
      popupMessage: {
        message: message,
        position: position
      }
    };
    expect(uiReducer(beforeActionState, togglePopupMessage)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, togglePopupMessage)).toMatchSnapshot();
  });
});

describe('REFRESH_ACTION_START_STATE', () => {
  const beforeActionState = {
    ...initialState,
    loader: {
      actions: [],
      refreshing: ['ACTION_NAME', 'ACTION_NAME', 'ACTION_NAME2']
    }
  };

  it('13returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(uiReducer(beforeActionState, action)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('14returns start action state', () => {
    const name = 'ACTION_NAME';
    const startAction = uiActions.startAction(name);
    const expectedState = {
      loader: {
        actions: [startAction.payload.action],
        refreshing: beforeActionState.loader.refreshing
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('15returns stop action state', () => {
    const name = 'ACTION_NAME';
    const stopAction = uiActions.stopAction(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: beforeActionState.loader.refreshing
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, stopAction)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, stopAction)).toMatchSnapshot();
  });

  it('16returns refresh action start state', () => {
    const name = 'ACTION_NAME';
    const refreshActionStart = uiActions.refreshActionStart(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: [...beforeActionState.loader.refreshing, refreshActionStart.payload.refreshAction]
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, refreshActionStart)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, refreshActionStart)).toMatchSnapshot();
  });

  it('17returns refresh action stop state', () => {
    const name = 'ACTION_NAME';
    const refreshActionStop = uiActions.refreshActionStop(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: ['ACTION_NAME2']
      },
      popupMessage: {
        message: null,
        position: null
      }
    };
    expect(uiReducer(beforeActionState, refreshActionStop)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, refreshActionStop)).toMatchSnapshot();
  });

  it('18returns toggle popup message state', () => {
    const message = 'message';
    const position = 'position';
    const togglePopupMessage = uiActions.togglePopupMessage(message, position);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: beforeActionState.loader.refreshing
      },
      popupMessage: {
        message: message,
        position: position
      }
    };
    expect(uiReducer(beforeActionState, togglePopupMessage)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, togglePopupMessage)).toMatchSnapshot();
  });
});

describe('TOGGLE_POPUP_MESSAGE_STATE', () => {
  const beforeActionState = {
    ...initialState,
    popupMessage: {
      message: 'message',
      position: 'position'
    }
  };

  it('19returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(uiReducer(beforeActionState, action)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('20returns start action state', () => {
    const name = 'ACTION_NAME';
    const startAction = uiActions.startAction(name);
    const expectedState = {
      loader: {
        actions: [startAction.payload.action],
        refreshing: []
      },
      popupMessage: {
        message: beforeActionState.popupMessage.message,
        position: beforeActionState.popupMessage.position
      }
    };
    expect(uiReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('21returns stop action state', () => {
    const name = 'ACTION_NAME';
    const stopAction = uiActions.stopAction(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: []
      },
      popupMessage: {
        message: beforeActionState.popupMessage.message,
        position: beforeActionState.popupMessage.position
      }
    };
    expect(uiReducer(beforeActionState, stopAction)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, stopAction)).toMatchSnapshot();
  });

  it('22returns refresh action start state', () => {
    const name = 'ACTION_NAME';
    const refreshActionStart = uiActions.refreshActionStart(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: [refreshActionStart.payload.refreshAction]
      },
      popupMessage: {
        message: beforeActionState.popupMessage.message,
        position: beforeActionState.popupMessage.position
      }
    };
    expect(uiReducer(beforeActionState, refreshActionStart)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, refreshActionStart)).toMatchSnapshot();
  });

  it('23returns refresh action stop state', () => {
    const name = 'ACTION_NAME';
    const refreshActionStop = uiActions.refreshActionStop(name);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: []
      },
      popupMessage: {
        message: beforeActionState.popupMessage.message,
        position: beforeActionState.popupMessage.position
      }
    };
    expect(uiReducer(beforeActionState, refreshActionStop)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, refreshActionStop)).toMatchSnapshot();
  });

  it('24returns toggle popup message state', () => {
    const message = 'message2';
    const position = 'position2';
    const togglePopupMessage = uiActions.togglePopupMessage(message, position);
    const expectedState = {
      loader: {
        actions: [],
        refreshing: beforeActionState.loader.refreshing
      },
      popupMessage: {
        message: togglePopupMessage.payload.message,
        position: togglePopupMessage.payload.position
      }
    };
    expect(uiReducer(beforeActionState, togglePopupMessage)).toEqual(expectedState);
    expect(uiReducer(beforeActionState, togglePopupMessage)).toMatchSnapshot();
  });
});
