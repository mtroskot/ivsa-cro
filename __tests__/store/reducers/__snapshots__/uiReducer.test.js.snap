// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL_STATE 1returns initial state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`INITIAL_STATE 2returns start action state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME",
        "params": undefined,
      },
    ],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`INITIAL_STATE 3returns stop action state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`INITIAL_STATE 4returns refresh action start state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [
      "ACTION_NAME",
    ],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`INITIAL_STATE 5returns refresh action stop state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`INITIAL_STATE 6returns toggle popup message state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": "message",
    "position": "position",
  },
}
`;

exports[`REFRESH_ACTION_START_STATE 13returns initial state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [
      "ACTION_NAME",
      "ACTION_NAME",
      "ACTION_NAME2",
    ],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`REFRESH_ACTION_START_STATE 14returns start action state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME",
        "params": undefined,
      },
    ],
    "refreshing": Array [
      "ACTION_NAME",
      "ACTION_NAME",
      "ACTION_NAME2",
    ],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`REFRESH_ACTION_START_STATE 15returns stop action state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [
      "ACTION_NAME",
      "ACTION_NAME",
      "ACTION_NAME2",
    ],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`REFRESH_ACTION_START_STATE 16returns refresh action start state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [
      "ACTION_NAME",
      "ACTION_NAME",
      "ACTION_NAME2",
      "ACTION_NAME",
    ],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`REFRESH_ACTION_START_STATE 17returns refresh action stop state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [
      "ACTION_NAME2",
    ],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`REFRESH_ACTION_START_STATE 18returns toggle popup message state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [
      "ACTION_NAME",
      "ACTION_NAME",
      "ACTION_NAME2",
    ],
  },
  "popupMessage": Object {
    "message": "message",
    "position": "position",
  },
}
`;

exports[`START_ACTION_STATE 7returns initial state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME2",
      },
    ],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`START_ACTION_STATE 8returns start action state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME2",
      },
      Object {
        "name": "ACTION_NAME",
        "params": undefined,
      },
    ],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`START_ACTION_STATE 9returns stop action state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME2",
      },
    ],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`START_ACTION_STATE 10returns refresh action start state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME2",
      },
    ],
    "refreshing": Array [
      "ACTION_NAME",
    ],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`START_ACTION_STATE 11returns refresh action stop state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME2",
      },
    ],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": null,
    "position": null,
  },
}
`;

exports[`START_ACTION_STATE 12returns toggle popup message state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME",
      },
      Object {
        "name": "ACTION_NAME2",
      },
    ],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": "message",
    "position": "position",
  },
}
`;

exports[`TOGGLE_POPUP_MESSAGE_STATE 19returns initial state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": "message",
    "position": "position",
  },
}
`;

exports[`TOGGLE_POPUP_MESSAGE_STATE 20returns start action state 1`] = `
Object {
  "loader": Object {
    "actions": Array [
      Object {
        "name": "ACTION_NAME",
        "params": undefined,
      },
    ],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": "message",
    "position": "position",
  },
}
`;

exports[`TOGGLE_POPUP_MESSAGE_STATE 21returns stop action state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": "message",
    "position": "position",
  },
}
`;

exports[`TOGGLE_POPUP_MESSAGE_STATE 22returns refresh action start state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [
      "ACTION_NAME",
    ],
  },
  "popupMessage": Object {
    "message": "message",
    "position": "position",
  },
}
`;

exports[`TOGGLE_POPUP_MESSAGE_STATE 23returns refresh action stop state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": "message",
    "position": "position",
  },
}
`;

exports[`TOGGLE_POPUP_MESSAGE_STATE 24returns toggle popup message state 1`] = `
Object {
  "loader": Object {
    "actions": Array [],
    "refreshing": Array [],
  },
  "popupMessage": Object {
    "message": "message2",
    "position": "position2",
  },
}
`;
